com.unity.netcode.gameobjects/TestHelpers/Runtime/NetcodeIntegrationTest.cs

```diff
@@ -757,6 +757,7 @@ protected IEnumerator StartClient(NetworkManager networkManager)
             {
                 // Wait for the new client to connect
                 yield return WaitForClientsConnectedOrTimeOut();
+                AssertOnTimeout($"{nameof(StartClient)} timed out waiting for all clients to be connected!\n {m_InternalErrorLog}");
```

Comment:
Nice catch on both of these

Reply:
Actually, I think you need to remove those... look just below... the client timing out is already being handled.

Reply:
Maybe remove the above assert on timeout and move the `OnNewClientStartedAndConnected(networkManager);` below 
like this?

```csharp
if (s_GlobalTimeoutHelper.TimedOut)
{
    AddRemoveNetworkManager(networkManager, false);
    Object.DestroyImmediate(networkManager.gameObject);
}
else
{
    OnNewClientStartedAndConnected(networkManager);
}
```